DOCSP-42281: vectorsearch options (#85)
10gen/docs-shared: dbx/jvm/v5.2-wn-items.rst

- When a sharded cluster operation is unsuccessful, the driver avoids selecting
  the same ``mongos`` server for operation retry attempts if other ``mongos``
  servers are available.

> - Separates the ``VectorSearchOptions`` type in previous versions into
>   the following types:

Reviewer: S: for clarity

```suggestion
- Replaces the ``VectorSearchOptions`` type with
  the following types:
```

  - ``ApproximateVectorSearchOptions``: Use this options parameter to

>     enable searches that may not return the exact closest vectors. You

Reviewer: I: sounds like this is possibility, not permission?

```suggestion
    enable searches that might not return the exact closest vectors. You
```

  the same ``mongos`` server for operation retry attempts if other ``mongos``
  servers are available.

- Enables exact vector search by replacing the ``VectorSearchOptions``

>   type with the following types:

Reviewer 2: The API for VectorSearchOptions has been changed by introducing more specific option subtypes for use
ApproximateVectorSearchOptions and ExactVectorSearchOptions.

Author: Will update the wording!
